northstar/src/runtime/token.rs

```diff
@@ -0,0 +1,123 @@
+use blake3::Hash;
+use lazy_static::lazy_static;
+use rand_core::{OsRng, RngCore};
+use std::io::{Cursor, Write};
+use std::ops::Add;
+use std::time::{Duration, Instant};
+use thiserror::Error;
+
+// Tokens are valid for one minute
+const TOKEN_EXPIRED_THRESHOLD: Duration = Duration::from_secs(60);
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Failed to write token")]
+    WriteToken,
+}
+
+lazy_static! {
+    static ref MAC_KEY: [u8; 32] = {
+        let mut key = [0u8; 32];
+        OsRng.fill_bytes(&mut key);
+        key
+    };
+    static ref FIRST_TOKEN_TIME: Instant = Instant::now();
+}
+
+pub type Timestamp = u64;
+
+pub struct Token {
+    time: Timestamp,
+    mac: blake3::Hash,
+}
+
+impl Token {
+    pub fn new(application: &str) -> Token {
```

Comment:
I'm struggling with the term `application` here. Maybe we can use something more generic here?

Reply:
How about 'usage' or general 'info'?